### PR TITLE
RD-1104 Refactor `manager` reducer

### PR DESCRIPTION
**Pull request by qooban**
## Description

Decided to implement RD-1104 before RD-3918. That should allow us to have not only easier to understand reducers hierarchy, but also better typing in places where changes for RD-3918 will be necessary.

Within this PR the following changes has been implemented:
* Added/Improved types for `license`, `cluster-status` and `tenants` resources
* Fixed reducers hierarchy around `managerReducer`
* Migrated few files to TypeScript

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/2333/) - PASSED (except known issue - RD-4666)

## Documentation
N/A

### CODE REVIEW

File: app/components/license/CurrentLicense.tsx [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+import type { LicenseResponse } from '../../../backend/routes/Auth.types';
+
+interface CurrentLicenseProps {
+    license?: LicenseResponse;
+}
 
-export default function CurrentLicense({ license }) {
-    if (_.isEmpty(license)) {
+const CurrentLicense: FunctionComponent<CurrentLicenseProps> = ({ license }) => {
+    if (!license || isEmpty(license)) {
```
Comment (kubama) [inline app/components/license/CurrentLicense.tsx:15]: Do we need `!license` check here?

File: app/components/sidebar/UserMenu.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ const UserMenu: FunctionComponent<SystemMenuGroupItemProps> = ({ expanded, onMod
     );
     const canChangePassword = useSelector((state: ReduxState) => !state.manager.isLdapEnabled);
     const tenantNames = useSelector((state: ReduxState) =>
-        state.manager.tenants.items.map((tenant: { name: string }) => tenant.name)
+        state.manager.tenants.items!.map((tenant: { name: string }) => tenant.name)
```
Comment (kubama) [inline app/components/sidebar/UserMenu.tsx:48]: Safer:
```suggestion
        state.manager.tenants.items?.map((tenant: { name: string }) => tenant.name)
```

File: app/reducers/managerReducer/licenseReducer.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+type ValueOf<T> = T[keyof T];
+export type LicenseStatus = ValueOf<typeof Consts.LICENSE>;
+
+export interface LicenseData {
+    data?: LicenseResponse;
+    isRequired?: boolean;
+    status?: LicenseStatus;
+}
+
+const license: Reducer<LicenseData> = (state = {}, action) => {
```
Comment (kubama) [inline app/reducers/managerReducer/licenseReducer.ts:16]: How about destructuring `action` properties?

File: app/reducers/managerReducer/licenseReducer.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import type { Reducer } from 'redux';
+import * as types from '../../actions/types';
+import Auth from '../../utils/auth';
+import Consts from '../../utils/consts';
+import type { LicenseResponse } from '../../../backend/routes/Auth.types';
+
+type ValueOf<T> = T[keyof T];
+export type LicenseStatus = ValueOf<typeof Consts.LICENSE>;
```
Comment (kubama) [inline app/reducers/managerReducer/licenseReducer.ts:8]: Cool! :slightly_smiling_face: :+1:

**Review by kubama — COMMENTED** [review]
Looks very good. I only have rather cosmetic comments.

File: app/components/license/CurrentLicense.tsx (hunk 1 above)
Comment (qooban) [reply app/components/license/CurrentLicense.tsx:15]: I added it to make TypeScript happy and do not throw errors "TS2532: Object is possibly 'undefined'." in 
L82: `const value = license[field.name];`
L96: `<Table.Cell>{field.format(license[field.name])}</Table.Cell>`

What do you think about that?

Comment (kubama) [reply app/components/license/CurrentLicense.tsx:15]: Now I see it, thanks :slightly_smiling_face: .

File: app/components/sidebar/UserMenu.tsx (hunk 2 above)
Comment (qooban) [reply app/components/sidebar/UserMenu.tsx:48]: I can do that, but then we'll need to add `!` in L120: `tenantNames={tenantNames!}` or refactor `ResetPagesModal`. What do you think?

File: app/reducers/managerReducer/licenseReducer.ts (hunk 3 above)
Comment (qooban) [reply app/reducers/managerReducer/licenseReducer.ts:16]: Sure, will do.

File: app/reducers/managerReducer/licenseReducer.ts (hunk 4 above)
Comment (qooban) [reply app/reducers/managerReducer/licenseReducer.ts:8]: Found it somewhere in StackOverflow and yeah, seems very useful 🙂 👍🏻

File: app/reducers/managerReducer/licenseReducer.ts (hunk 3 above)
Comment (qooban) [reply app/reducers/managerReducer/licenseReducer.ts]: Done.

File: app/components/sidebar/UserMenu.tsx (hunk 2 above)
Comment (kubama) [reply app/components/sidebar/UserMenu.tsx:48]: Ok, we can leave it then.

**Review by kubama — APPROVED** [review]

**Pull request merged**